source/base/quadrature.cc

```diff
@@ -26,11 +26,31 @@
 DEAL_II_NAMESPACE_OPEN
 
 
+namespace
+{
+  template <int dim>
+  void
+  set_tensor_basis_1d(const Quadrature<dim> &,
+                      std::vector<Quadrature<1>> &)
+  {}
```

Comment:
Derived classes typically set quadrature points and weights after the constructor for the base class is called in the body of the constructor. This prevents `tensor_basis` from being `const`.